src/logic/main.js

```diff
@@ -51,11 +54,101 @@ function isMoveLegal() {
+    else if ( (actionState === `${CardActionEnum.Taki}Invoked`) &&
+            (matchedCard = getCardInHand(bot, [{color: leadingCard.color}])) ){
+            selectedCard = matchedCard;
+    } // ( if we got here no actionState was invoked)
+    else
+    {
+        // 4.2 if bot has a twoPlus card  with the same color as the leadingCard
+        // or the leadingCard is a non-active twoPlus - mark it as selectedCard.
+        if ( (matchedCard = getCardInHand(bot, [{action: CardActionEnum.TwoPlus}, {color: leadingCard.color}])) ||
+             ( (leadingCard.action === CardActionEnum.TwoPlus) && (matchedCard = getCardInHand(bot, [{action: CardActionEnum.TwoPlus}]) )) ) {
+            selectedCard = matchedCard ;
+        }
+        // 4.3 if bot has ChangeColor card and you're allowed to put it (actionState is none)- mark it as selectedCard.
+        else if (matchedCard = getCardInHand(bot, [{action: CardActionEnum.ChangeColor}])) {
+            selectedCard = matchedCard;
+        }
+        // 4.4 if bot has  a Stop card with the same color as the leadingCard - mark it as selectedCard.
+        else if (matchedCard = getCardInHand(bot, [{action: CardActionEnum.Stop}, {color: leadingCard.color}])) {
+            selectedCard = matchedCard;
+        }
+        // 4.5 if bot has a Plus card with the same color as the leadingCard - mark it as selectedCard.
+        else if (matchedCard = getCardInHand(bot, [{action: CardActionEnum.Plus}, {color: leadingCard.color}])) {
+            selectedCard = matchedCard;
+        }
+        // 4.6 if bot has a superTaki card - mark it as selectedCard.
+        else if (matchedCard = getCardInHand(bot, [{action: CardActionEnum.SuperTaki}])) {
+            selectedCard = matchedCard;
+        }
+        // 4.7 if bot has a taki card with the same color as the leadingCard - mark it as the selectedCard.
+        else if (matchedCard = getCardInHand(bot, [{action: CardActionEnum.Taki}, {color: leadingCard.color}])) {
+            selectedCard = matchedCard;
+        }
+        // 4.8 if you have a card with the same color as the leading card - mark it as the selectedCard.
+        else if (matchedCard = getCardInHand(bot, [{color: leadingCard.color}])) {
+            selectedCard = matchedCard;
+        }
+        // 4.9 if you have a card with the same number as the leading card - mark it as the selectedCard.
+        else if ((leadingCard.number !== null) &&
+            (matchedCard = getCardInHand(bot, [{number: leadingCard.number}]))) {
+            selectedCard = matchedCard;
+        }
+        // 4.10 if none of the conditions above happen - mark the top card of the draw pile as the selectedCard.
+        else {
+            debugger;
+            selectedCard = GameState.DrawPile.getTop();
+        }
+    }
+    // todo : try to see if i could use the "selectedCard = matchedCard " line only once in this function
```

Comment:
Please delete todo here and open a technical story to improve this: selectedCard = matchedCard